alns/stop/tests/test_no_improvement.py

```diff
@@ -0,0 +1,92 @@
+import pytest
+from numpy.random import RandomState
+from numpy.testing import assert_, assert_equal, assert_raises
+
+from alns.tests.states import One, Zero, Two
+
+from alns.stop import NoImprovement
+
+
+@pytest.mark.parametrize("max_iterations", [-10, -100, -1000])
+def test_raise_negative_parameters(max_iterations: int):
+    """
+    max_iterations cannot be negative.
+    """
+    with assert_raises(ValueError):
+        NoImprovement(max_iterations)
+
+
+@pytest.mark.parametrize("max_iterations", [0, 10, 100, 1000])
+def test_does_not_raise(max_iterations: int):
+    """
+    Non-negative integers should not raise.
+    """
+    NoImprovement(max_iterations)
+
+
+def test_max_iterations():
+    """
+    Test if the max_iterations parameter is correctly set.
+    """
+    stop = NoImprovement(3)
+    assert_equal(stop.max_iterations, 3)
```

Comment:
Idea: parametrize this as well.

Since this is functionally the same as the `does_not_raise` test above, maybe remove that test and use this instead?

Reply:
Question: when does it help to parametrize tests? I initially had them, but then removed them again. I thought that it was not necessary as long as the chosen parameters don't touch the edge cases. The edge cases are tested separately.

Reply:
In this particular instance I just wanted to test that we get the initial input back from max_iterations, for some different values.

Normally I parametrise tests with tuples of `(*inputs, output)`. That's useful to cover all sorts of edge cases with a single test!